src/main/java/com/es/phoneshop/service/impl/DefaultCartService.java

```diff
@@ -73,4 +68,10 @@ public synchronized void add(Cart cart, Long productId, int quantity) throws Out
             readWriteLock.writeLock().unlock();
         }
     }
+    private Optional<CartItem> findCartItem(Cart cart, Product product) {
+        return cart.getItems()
+                .stream()
+                .filter(c -> c.getProduct().equals(product))
```

Comment:
equals by productCode.
Or implement equals for Product.java(this is a worse option)